### PR TITLE
Limit the number of concurrent coroutines

### PR DESCRIPTION
**Pull request by vondele**
reinstates a limit on the number of concurrent coroutines, while better than threads, they still use too much memory to be used unbounded.

### CODE REVIEW

File: cdbsearch.py [hunk 1]
```diff
@@ -411,41 +413,46 @@ async def search(self, board, depth):
         tasks = {}
         tried_unscored = False
 
-        for move in board.legal_moves:
-            ucimove = move.uci()
-            score = scored_db_moves.get(ucimove, None)
-            newdepth = self.move_depth(bestscore, worstscore, score, depth)
+        # guarantee sufficient length of the executorTree list
```
Comment (robertnurnberg) [inline cdbsearch.py:416]: comment should read `semaphoreTree`

Comment (robertnurnberg) [reply cdbsearch.py:416]: in addition, comment could say something like: "limit concurrent tasks per level of the search tree" or similar?

**Comment by robertnurnberg** [comment]
Tested locally without any problems, looks good to merge.

**Comment by robertnurnberg** [comment]
I do not know much about multithreading etc, so I am just curious to what is the rationale behind choosing the limit in

```python
self.semaphoreTree.append(asyncio.Semaphore(2 * self.concurrency))
```

Would there be any performance gain by choosing `4 * self.concurrency`, say?

**Comment by vondele** [comment]
I don't know for certain, but I don't think so. I think it is either the limit set to the concurrent access of the cdb api, or the time spent in waiting for it after enqueuing a position, or the time spent in just python code traversing the tree (i.e. 100% cpu load). I probably will add a counter more to the code to get some understanding of how much time is spent waiting in queryall.

**Comment by robertnurnberg** [comment]
Ok thanks. Might be worth checking out dubslow's repo. I think he manages to have lightning fast API access.

**Pull request merged**